File: 48. Rotate Image.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: void Do not return anything, modify matrix in-place instead.
        """
        if len(matrix)>1:
            matrix.reverse()
            for i in xrange(len(matrix)):
                for j in xrange(i+1, len(matrix[0])):
                    matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]

File: 48. Rotate Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        if (matrix.size() > 1)
        {
            reverse(matrix.begin(), matrix.end());
            for (int i = 0; i < matrix.size(); i++)
            {
                for (int j = i + 1; j < matrix[0].size(); j++)
                {
                    int temp = matrix[i][j];
                    matrix[i][j] = matrix[j][i];
                    matrix[j][i] = temp;
                }
            }
        }
    }
};
